spec/java/util/stream/DoubleStream.main.lsl

```diff
@@ -1022,54 +1020,66 @@ automaton DoubleStreamAutomaton
         }
         else
         {
+            action ASSUME(this.length > 0);
+
             var dropLength: int = 0;
 
             var i: int = 0;
-            action LOOP_FOR(
-                i, 0, this.length, +1,
+            action LOOP_WHILE(
+                i < this.length && action CALL(predicate, [this.storage[i]]),
                 _dropWhile_loop(i, dropLength, predicate)
             );
 
-            val newLength: int = this.length - dropLength;
-            val newStorage: array<double> = action ARRAY_NEW("double", newLength);
-
-            var j: int = 0;
-            action LOOP_FOR(
-                i, dropLength, this.length, +1,
-                _copy_dropWhile_loop(i, j, newStorage)
-            );
-
-            result = new DoubleStreamAutomaton(state = Initialized,
-                storage = newStorage,
-                length = newLength,
-                closeHandlers = this.closeHandlers,
-            );
+            if (dropLength == 0)
+            {
+                result = new DoubleStreamAutomaton(state = Initialized,
+                    storage = this.storage,
+                    length = this.length,
+                    closeHandlers = this.closeHandlers,
+                );
+            }
+            else
+            {
+                val newLength: int = this.length - dropLength;
+                val newStorage: array<int> = action ARRAY_NEW("int", newLength);
```

Comment:
Invalid storage element type: expecting `double` instead of `int`.

Reply:
Fixed;